### PR TITLE
[FIX #4327] Migrate to react-native-firebase

### LINKED ISSUE #4327: Migrate from `react-native-fcm` to `react-native-firebase`
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a developer, I want to upgrade from `react-native-fcm` to `react-native-firebase` so that I have access to the latest APIs and bugfixes.

The `react-native-fcm` package is no longer guaranteed to have any active development, while `react-native-firebase` has great backing:

image

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

### Solution
[comment]: # (Please summarise the solution and provide a task list on what needs to be fixed.)
*Summary*: Follow steps in https://github.com/evollu/react-native-fcm#to-existing-react-native-fcm-users to upgrade current setup to `react-native-firebase`

### PR DESCRIPTION
**Pull request by georgiemathews**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

Fixes #4327

### Summary:

[comment]: # (Summarise the problem and how the pull request solves it)
Migrates Notifications to use invertase/react-native-firebase

### Steps to test:
- Open Status
- Accept Notifications Request
- See if notifications are received

status: ready

### CODE REVIEW

**Comment by pedropombeiro** [comment]
Thanks for the PR @georgiemathews!

The build is currently failing in Jenkins:

image

File: android/app/build.gradle [hunk 1]
```diff
@@ -207,7 +208,7 @@ dependencies {
     implementation "com.android.support:appcompat-v7:26.1.0"
     // Force using exact RN version instead of relying on gradle dependency resolution
     // https://docs.gradle.org/current/userguide/introduction_dependency_management.html#sec:dependency_resolution
-    compile ("com.facebook.react:react-native:0.53.3") { force = true }  // From node_modules
+    implementation ("com.facebook.react:react-native:0.53.3") { force = true }  // From node_modules
```
Comment (jeluard) [inline android/app/build.gradle]: Why this change?

File: android/app/build.gradle [hunk 2]
```diff
@@ -216,9 +217,11 @@ dependencies {
...
     implementation project(':react-native-webview-bridge')
     implementation project(':react-native-config')
-    implementation project(':react-native-fcm')
-    implementation 'com.google.firebase:firebase-core:10.0.1' //this decides your firebase SDK version
-    compile project(':react-native-testfairy')
+    implementation 'com.google.android.gms:play-services-base:15.0.0'
+    implementation 'com.google.firebase:firebase-core:15.0.2' //this decides your firebase SDK version
+    implementation "com.google.firebase:firebase-messaging:15.0.2"
+    implementation 'me.leolin:ShortcutBadger:1.1.21@aar'
+    implementation project(':react-native-testfairy')
```
Comment (jeluard) [inline android/app/build.gradle]: Please comment why we need all those.

File: android/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -9,6 +9,8 @@
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
     <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
     <uses-permission android:name="android.permission.READ_PROFILE" />
+    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
```
Comment (jeluard) [inline android/app/src/main/AndroidManifest.xml:12]: This require comment too.

File: android/app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -33,16 +35,16 @@
...
+        android:resource="@drawable/ic_stat_ic_notification" />
         <activity
                 android:name=".MainActivity"
                 android:label="@string/app_name"
                 android:screenOrientation="portrait"
                 android:windowSoftInputMode="adjustResize"
                 android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
                 android:resizeableActivity="false"
-                android:launchMode="singleTask">
+                android:launchMode="singleTop">
```
Comment (jeluard) [inline android/app/src/main/AndroidManifest.xml]: What is this?

File: android/app/src/main/java/im/status/ethereum/MainActivity.java [hunk 5]
```diff
@@ -94,13 +94,6 @@ private Intent createNotificationSettingsIntent() {
         return intent;
     }
 
-    @Override
-    public void onNewIntent(final Intent intent) {
-        if (intent.getDataString() != null && intent.getData().getScheme().startsWith("app-settings")) {
-          startActivity(createNotificationSettingsIntent());
```
Comment (jeluard) [inline android/app/src/main/java/im/status/ethereum/MainActivity.java]: Is this related?

File: android/build.gradle [hunk 6]
```diff
@@ -6,8 +6,8 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.0.1'
-        classpath 'com.google.gms:google-services:3.0.0'
+        classpath 'com.android.tools.build:gradle:3.1.2'
+        classpath 'com.google.gms:google-services:3.2.1'
```
Comment (jeluard) [inline android/build.gradle:10]: Is this required?

File: android/build.gradle [hunk 7]
```diff
@@ -25,5 +24,6 @@ allprojects {
         flatDir { dirs 'libs' }
         maven { url "http://139.162.11.12:8081/artifactory/libs-release-local" }
         maven { url "https://jitpack.io" }
+        google()
```
Comment (jeluard) [inline android/build.gradle:27]: ?

**Review by jeluard — CHANGES_REQUESTED** [review]
Lots of changes look unrelated with the issue. Could we reduce the scope of this PR?

File: android/app/build.gradle (hunk 1 above)
Comment (georgiemathews) [reply android/app/build.gradle]: image

Comment (jeluard) [reply android/app/build.gradle]: @dmitryn WDYT? Looks like it's reverting your recent change?

File: android/build.gradle (hunk 7 above)
Comment (georgiemathews) [reply android/build.gradle]: image

File: android/build.gradle (hunk 6 above)
Comment (georgiemathews) [reply android/build.gradle:10]: image
image

File: android/app/src/main/java/im/status/ethereum/MainActivity.java (hunk 5 above)
Comment (georgiemathews) [reply android/app/src/main/java/im/status/ethereum/MainActivity.java]: this is from the react-native-fcm documentation
screen shot 2018-05-25 at 7 58 04 am

Comment (jeluard) [reply android/app/src/main/java/im/status/ethereum/MainActivity.java]: Well that's not the same thing you removed?

File: android/build.gradle (hunk 7 above)
Comment (jeluard) [reply android/build.gradle:27]: It was already there. Any reason to change its place?

File: android/app/src/main/java/im/status/ethereum/MainActivity.java (hunk 5 above)
Comment (georgiemathews) [reply android/app/src/main/java/im/status/ethereum/MainActivity.java]: @jeluard thanks for the catch, I see this was implemented for easy access to the settings from profile. Thanks, will revert

File: android/build.gradle (hunk 7 above)
Comment (georgiemathews) [reply android/build.gradle:27]: image

File: android/app/src/main/AndroidManifest.xml (hunk 4 above)
Comment (georgiemathews) [reply android/app/src/main/AndroidManifest.xml]: This was recommended in the react-native-firebase, but I've looked into it and singleTask will functionally work as well. I'll revert this

File: android/app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (georgiemathews) [reply android/app/src/main/AndroidManifest.xml:12]: This is required for the notifications 
image

File: android/app/build.gradle (hunk 2 above)
Comment (georgiemathews) [reply android/app/build.gradle]: image
image
shortcut badger allows for badges on android

Comment (georgiemathews) [reply android/app/build.gradle]: image

Comment (jeluard) [reply android/app/build.gradle]: badges? Don't think we want that.

Comment (georgiemathews) [reply android/app/build.gradle]: okay @jeluard, will revert that as well.

File: ios/Podfile [hunk 8]
```diff
@@ -11,13 +11,14 @@
 #pod 'Firebase/Messaging', '3.17.0'
 
 target 'StatusIm' do
-  platform :ios, '8.0'
+  platform :ios, '9.0'
```
Comment (jeluard) [inline ios/Podfile]: What does that mean with regards for supported mobiles?

File: ios/Podfile.lock [hunk 9]
```diff
@@ -1,57 +1,90 @@
...
+  React: af66244a7d630be2aef03b4ddfe3570e4eff39fc
+  react-native-background-timer: 63dcbf37dbcf294b5c6c071afcdc661fa06a7594
+  RNKeychain: 627c6095cef215dd3d9804a9a9cf45ab96aa3997
+  yoga: 55da126afc384965b96bff46652464373b330add
 
-PODFILE CHECKSUM: 653ad2d673bf8aa8127325a078e07aedc8dbe463
+PODFILE CHECKSUM: c51578b58da3d2ed7f180c1ec7ec0ce660a03aaa
 
-COCOAPODS: 1.3.1
+COCOAPODS: 1.5.2
```
Comment (jeluard) [inline ios/Podfile.lock]: Is this also required?

File: package.json [hunk 10]
```diff
@@ -39,8 +39,8 @@
     "react-native-config": "0.9.0",
     "react-native-crypto": "2.1.1",
     "react-native-dialogs": "0.0.20",
-    "react-native-fcm": "10.0.3",
     "react-native-fetch-polyfill": "1.1.2",
+    "react-native-firebase": "^4.2.0",
```
Comment (jeluard) [inline package.json]: We want fixed versions.

File: android/app/src/main/AndroidManifest.xml [hunk 11]
```diff
@@ -33,8 +35,8 @@
             android:largeHeap="true">
       <meta-data android:name="firebase_analytics_collection_deactivated" android:value="true" />
       <meta-data
-              android:name="com.google.firebase.messaging.default_notification_icon"
-              android:resource="@drawable/ic_stat_status_notification" />
+        android:name="com.google.firebase.messaging.default_notification_icon"
+        android:resource="@drawable/ic_stat_ic_notification" />
```
Comment (pedropombeiro) [inline android/app/src/main/AndroidManifest.xml]: I'm getting a build error regarding this resource:
```
 AAPT: No resource found that matches the given name (at 'resource' with value '@drawable/ic_stat_ic_notification').
```

Is this change intentional?

**Comment by pedropombeiro** [comment]
@georgiemathews I'm seeing this when attempting to run your branch. Any ideas?

image

**Comment by georgiemathews** [comment]
@PombeirP should be fixed now!

**Comment by pedropombeiro** [comment]
Confirmed that app is able to send PNs on this branch. Could you please rebase on latest `develop` branch @georgiemathews?

**Comment by pedropombeiro** [comment]
@georgiemathews there must have been a problem in the new rebase, it rolls back the upgrade in a2762eb19e45810184671b6c47f11faff0270124. I believe it also removed `babel-polyfill` from `package-lock.json`, not sure if that is intended.
